v2/protocol/http/protocol.go

```diff
@@ -223,34 +224,45 @@ func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	done := make(chan struct{})
-	var finishErr error
 
+	var finishErr error
 	m.OnFinish = func(err error) error {
 		finishErr = err
 		return nil
 	}
 
-	var fn protocol.ResponseFn = func(ctx context.Context, resp binding.Message, er protocol.Result, transformers ...binding.Transformer) error {
+	var fn protocol.ResponseFn = func(ctx context.Context, respMsg binding.Message, res protocol.Result, transformers ...binding.Transformer) error {
 		// Unblock the ServeHTTP after the reply is written
 		defer func() {
 			done <- struct{}{}
 		}()
-		status := http.StatusOK
+
 		if finishErr != nil {
-			http.Error(rw, fmt.Sprintf("cannot forward CloudEvent: %v", finishErr), http.StatusInternalServerError)
```

Comment:
@n3wscott I've got a feeling we want to return here, can you confirm?

Reply:
if finishErr is set it means that the message is closed and responding will do nothing, so I think that logic makes sense. But also, because the message is closed, the http Error will likely be ignored by the caller.